File: 03-march/27-matrix-similarity-after-cyclic-shifts/solution.java
class Solution {
    public boolean areSimilar(int[][] mat, int k) {
        int n = mat[0].length;
        int shift = k % n;
        if (shift == 0) return true;
        for (int i = 0; i < mat.length; i++) {
            for (int j = 0; j < n; j++) {
                int shifted = (i % 2 == 0)
                    ? mat[i][(j + shift) % n]
                    : mat[i][(j - shift + n) % n];
                if (mat[i][j] != shifted) return false;
            }
        }
        return true;
    }
}

File: 03-march/27-matrix-similarity-after-cyclic-shifts/solution.py
class Solution:
    def areSimilar(self, mat: List[List[int]], k: int) -> bool:
        n = len(mat[0])
        for i, row in enumerate(mat):
            shift = k % n
            if shift == 0:
                continue
            if i % 2 == 0:
                shifted = row[shift:] + row[:shift]
            else:
                shifted = row[n - shift:] + row[:n - shift]
            if shifted != row:
                return False
        return True
